aws/resource_aws_s3_bucket.go

```diff
@@ -2191,33 +2186,38 @@ func resourceAwsS3BucketLifecycleUpdate(s3conn *s3.S3, d *schema.ResourceData) e
 		}
 
 		// Expiration
-		expiration := d.Get(fmt.Sprintf("lifecycle_rule.%d.expiration", i)).(*schema.Set).List()
+		expiration := d.Get(fmt.Sprintf("lifecycle_rule.%d.expiration", i)).([]interface{})
```

Comment:
wondering if this is the correct way to get the `expiration` value from the resource b/c I noticed if expiration is passed in as empty `expiration {}`  block for example in a TF config, the value of this `expiration` variable is `[ nil ]`, requiring the 2 `if` checks below..while previously only 1 `if` was needed

Reply:
value extracted here and in `nc_expiration` at L2212 could also explain new test failure seen in ` TestAccAWSS3Bucket_LifecycleRule_Expiration_EmptyConfigurationBlock`...though following the logic, the error seems to makes sense to me as nothing in the `rule` object will be populated in terms of an `action` if one were to pass just the `enabled` and `id` params to `lifecycle_rule`